services/crud-apis/utils/config.py

```diff
@@ -0,0 +1,23 @@
+import os
+
+from dotenv import load_dotenv
+from pydantic import BaseModel
+from pydantic_settings import BaseSettings
+
+load_dotenv()
+
+
+class Settings(BaseSettings):
+    env: str = "local"
+    workspace: str | None = None
+    endpoint_url: str | None = None
+
+    def model_post_init(self, __context: BaseModel) -> None:
+        super().model_post_init(__context)
+        self.env = os.getenv("ENVIRONMENT", self.env)
+        self.workspace = os.getenv("WORKSPACE", self.workspace)
+        self.endpoint_url = os.getenv("ENDPOINT_URL", self.endpoint_url)
+
+
+def get_env_variables() -> Settings:
+    return Settings()
```

Comment:
This can be simplified, I'd also just directly initialise `Settings` when you need it (you can add cached fetching from SSM and secrets manager if we need to)

```suggestion
import os

from pydantic import BaseModel, Field
from pydantic_settings import BaseSettings, SettingsConfigDict


class Settings(BaseSettings):
    model_config = SettingsConfigDict(env_file=".env", env_file_encoding="utf-8")

    env: str = Field("local", alias="ENVIRONMENT")
    workspace: str | None = Field(None, alias="WORKSPACE")
    endpoint_url: str | None = Field(None, alias="ENDPOINT_URL")

```